pkg/contract_watcher/header/transformer/transformer.go

```diff
@@ -248,20 +252,19 @@ func (tr *Transformer) Execute() error {
 			continue
 		}
 
-		// Sort logs by the contract they belong to
 		for _, log := range allLogs {
 			addr := strings.ToLower(log.Address.Hex())
 			sortedLogs[addr] = append(sortedLogs[addr], log)
 		}
 
 		// Process logs for each contract
 		for conAddr, logs := range sortedLogs {
-			if logs == nil {
+			con := tr.Contracts[conAddr]
+			if len(logs) < 1 {
```

Comment:
Any potential consequences with removing the `nil` check?

Reply:
The `len(logs)` substitutes for it, but there is no case where the logs will be length 0 without being `nil` so I can revert it back.

Reply:
Huh, TIL! Thought that'd lead to a NPE :)